projects/vdk-core/src/vdk/internal/builtin_plugins/run/file_based_step.py

```diff
@@ -133,7 +134,21 @@ def invoke_run_function(func: Callable, job_input: IJobInput):
             if arg_name in parameter_names
         }
         if actual_arguments:
-            func(**actual_arguments)
+            try:
+                func(**actual_arguments)
+            except BaseException as e:
```

Comment:
Why not catching `Exception`?

Reply:
To be able to classy properly things lie All Errors - including things like GeneratorExit, MemoryError, etc.

Reply:
Actually only GenertorExit,SystemExit and KeyboardInterrupt inherit BaseException (from the built in exceptions). If user code raises them how should we handle them?

Reply:
users will do whatever they want. 

Here we catch it, pass it through classifier to detect the type of error and then re-raise it wrap in the appropriate VDK exception. 

The question is do we want to handle GenertorExit,SystemExit and KeyboardInterrupt the same way?